pkg/gh.go

```diff
@@ -15,5 +16,17 @@ func GithubClient() *github.Client {
 		httpclient = oauth2.NewClient(context.Background(), oauth2.StaticTokenSource(&oauth2.Token{AccessToken: token}))
 	}
 	githubClient := github.NewClient(httpclient)
+	rateLimit, _, err := githubClient.RateLimits(context.Background())
+	if err != nil {
+		gologger.Error().Msgf("error while getting rate limits %s ", err.Error())
+
+	}
+	if rateLimit.Core.Remaining <= 0 {
+		gologger.Error().Msgf("error for remaining request per hour: %s", err.Error())
+		if arlErr, ok := err.(*github.AbuseRateLimitError); ok {
```

Comment:
This casting should be moved here https://github.com/projectdiscovery/pdtm/blob/034287ab34c47e325f1edb45a5b7fdac54c8d9be/pkg/install.go#L80 if we want to provide more informative message about rate limit. Internally github client library already returns `github.AbuseRateLimitError` without additional requests.